sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -494,14 +525,15 @@ public BigtableService apply(PipelineOptions options) {
     @Override
     public void validate(PCollection<KV<ByteString, Iterable<Mutation>>> input) {
       checkArgument(options != null, "BigtableOptions not specified");
-      checkArgument(!tableId.isEmpty(), "Table ID not specified");
+      if(tableId == null) throw new IllegalArgumentException("Table ID not specified");
+      checkArgument(!tableId.get().isEmpty(), "Table ID not specified");
       try {
         checkArgument(
-            getBigtableService(input.getPipeline().getOptions()).tableExists(tableId),
+            getBigtableService(input.getPipeline().getOptions()).tableExists(tableId.get()),
             "Table %s does not exist",
             tableId);
       } catch (IOException e) {
-        LOG.warn("Error checking whether table {} exists; proceeding.", tableId, e);
+        LOG.warn("Error checking whether table {} exists; proceeding.", ValueProvider.StaticValueProvider.of(tableId), e);
```

Comment:
I don't think this wrapping of `tableId` into a StaticValueProvider is necessary (or desirable) for logging.